hack/generate.sh

```diff
@@ -36,20 +37,42 @@ virtapi_version=$(getVersion ".VirtApiSha")
 virtcontroller_version=$(getVersion ".VirtControllerSha")
 virthandler_version=$(getVersion ".VirtHandlerSha")
 virtlauncher_version=$(getVersion ".VirtLauncherSha")
+virtoperator_version=$(getVersion ".VirtOperatorSha")
 
 # used as env var for operator
 function getShasum() {
     echo "{{if $1}}@{{$1}}{{end}}"
 }
+
+# without the '@' symbole used in 'getShasum'
+function getRawShasum() {
+    echo "{{if $1}}{{$1}}{{end}}"
+}
 virtapi_sha=$(getShasum ".VirtApiSha")
 virtcontroller_sha=$(getShasum ".VirtControllerSha")
 virthandler_sha=$(getShasum ".VirtHandlerSha")
 virtlauncher_sha=$(getShasum ".VirtLauncherSha")
 
+virtapi_rawsha=$(getRawShasum ".VirtApiSha")
+virtcontroller_rawsha=$(getRawShasum ".VirtControllerSha")
+virthandler_rawsha=$(getRawShasum ".VirtHandlerSha")
+virtlauncher_rawsha=$(getRawShasum ".VirtLauncherSha")
+
 ${KUBEVIRT_DIR}/tools/resource-generator/resource-generator --type=virt-api --namespace={{.Namespace}} --repository={{.DockerPrefix}} --version="$virtapi_version" --pullPolicy={{.ImagePullPolicy}} --verbosity={{.Verbosity}} >${KUBEVIRT_DIR}/manifests/generated/virt-api.yaml.in
 ${KUBEVIRT_DIR}/tools/resource-generator/resource-generator --type=virt-controller --namespace={{.Namespace}} --repository={{.DockerPrefix}} --version="$virtcontroller_version" --launcherVersion="$virtlauncher_version" --pullPolicy={{.ImagePullPolicy}} --verbosity={{.Verbosity}} >${KUBEVIRT_DIR}/manifests/generated/virt-controller.yaml.in
 ${KUBEVIRT_DIR}/tools/resource-generator/resource-generator --type=virt-handler --namespace={{.Namespace}} --repository={{.DockerPrefix}} --version="$virthandler_version" --pullPolicy={{.ImagePullPolicy}} --verbosity={{.Verbosity}} >${KUBEVIRT_DIR}/manifests/generated/virt-handler.yaml.in
 
+# The generation code for CSV requires a valid semver to be used.
+# But we're trying to generate a template for a CSV here from code
```

Comment:
if so, maybe you should call this tool `csv-template-generator` ?

Reply:
well, not exactly.  This code comment is in relation only to how KubeVirt is utilizing the tool in our build flow. I'm referring to something here that's just specific to KubeVirt and not HCO's usage of csv-generator. 

KubeVirt is generating a template using the tool by replacing the arguments passed into the csv-generator with go template variables... So instead of passing in the actual operator's image sha, we pass in something like this `{.OperatorSha}`, which is later filled in using another build step. 

The HCO will actually be passing in real args to csv-generator though